labelbox/schema/bulk_import_request.py

```diff
@@ -318,3 +203,115 @@ def __build_bulk_import_request_from_result(
+            projectId: $projectId,
+            name: $name,
+            fileUrl: $fileUrl
+        }) {
+            %s
+        }
+    }
+    """ % _build_results_query_part()
+    params = {"projectId": project_id, "name": name, "fileUrl": url}
+    bulk_import_request_response = client.execute(query_str, params=params)
+    return BulkImportRequest.from_result(
+        client, bulk_import_request_response["createBulkImportRequest"])
+
+
+def create_from_objects(cls, client: Client, project_id: str, name: str,
+                        predictions: Iterable[dict]) -> 'BulkImportRequest':
+    """
+    Creates a BulkImportRequest from an iterable of dictionaries conforming to
+    JSON predictions format, e.g.:
+    ``{
+        "uuid": "9fd9a92e-2560-4e77-81d4-b2e955800092",
+        "schemaId": "ckappz7d700gn0zbocmqkwd9i",
+        "dataRow": {
+            "id": "ck1s02fqxm8fi0757f0e6qtdc"
+        },
+        "bbox": {
+            "top": 48,
+            "left": 58,
+            "height": 865,
+            "width": 1512
+        }
+    }``
+
+    Args:
+        client (Client): a Labelbox client
+        project_id (str): id of project for which predictions will be imported
+        name (str): name of BulkImportRequest
+        predictions (Iterable[dict]): iterable of dictionaries representing predictions
+    Returns:
+        BulkImportRequest object
+    """
+    data_str = ndjson.dumps(predictions)
+    data = data_str.encode('utf-8')
+    file_name = __make_file_name(project_id, name)
+    request_data = __make_request_data(project_id, name, len(data_str),
+                                       file_name)
+    file_data = (file_name, data, NDJSON_MIME_TYPE)
+    response_data = __send_create_file_command(client, request_data, file_name,
+                                               file_data)
+    return BulkImportRequest.from_result(
+        client, response_data["createBulkImportRequest"])
+
+
+def create_from_local_file(cls,
+                           client: Client,
+                           project_id: str,
+                           name: str,
+                           file: Path,
+                           validate_file=True) -> 'BulkImportRequest':
+    """
+    Creates a BulkImportRequest from a local ndjson file with predictions.
+
+    Args:
+        client (Client): a Labelbox client
+        project_id (str): id of project for which predictions will be imported
+        name (str): name of BulkImportRequest
+        file (Path): local ndjson file with predictions
+        validate_file (bool): a flag indicating if there should be a validation
+            if `file` is a valid ndjson file
+    Returns:
+        BulkImportRequest object
+    """
+    file_name = __make_file_name(project_id, name)
+    content_length = file.stat().st_size
+    request_data = __make_request_data(project_id, name, content_length,
+                                       file_name)
+    if validate_file:
```

Comment:
@gszpak ptal at the changes here

this way we don't read into memory completely for validation